modules/gdscript/gdscript_parser.cpp

```diff
@@ -3852,11 +3855,53 @@ bool GDScriptParser::validate_annotation_arguments(AnnotationNode *p_annotation)
 		p_annotation->resolved_arguments.push_back(value);
 	}
 
+	if (p_annotation->name == SNAME("@uid")) {
+		ExpressionNode *argument = p_annotation->arguments[0];
+
+		if (argument->type != Node::LITERAL) {
+			push_error(R"(Argument 1 of annotation "@uid" must be a string literal.)", argument);
+			return false;
+		}
+
+		Variant value = static_cast<LiteralNode *>(argument)->value;
+
+		if (value.get_type() != Variant::STRING) {
+			push_error(R"(Argument 1 of annotation "@uid" must be a string literal.)", argument);
+			return false;
+		}
+
+		p_annotation->resolved_arguments.push_back(value);
+	}
+
```

Comment:
This can be combined with the `@icon` above, just replace `@icon` with `p_annotation->name` in `push_error()`.

```c++
push_error(vformat(R"(Argument 1 of annotation "%s" must be a string literal.)", p_annotation->name), argument);
```